### Leitura do arquivo .pkl

Leitura do arquivo onde foi salvo os previsores e as classes com seu devido pre processsamento

In [7]:
import pickle

In [8]:
with open('../../variaveis_preprocessadas/census.pkl', 'rb') as f:
  previsores_treinamento, classe_treinamento, previsores_teste, classe_teste = pickle.load(f)


### Utilização do algoritmo Àrvores de decisão

In [9]:
from sklearn.tree import DecisionTreeClassifier

tree = DecisionTreeClassifier()
tree.fit(previsores_treinamento, classe_treinamento)

previsoes = tree.predict(previsores_teste)

### Avaliação dos resultados do algoritmo Àrvores de decisão

In [10]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [11]:
score = accuracy_score(classe_teste, previsoes)
score

0.8096212896622313

In [12]:
confusion_matrix(classe_teste, previsoes)

array([[3204,  489],
       [ 441,  751]], dtype=int64)

In [13]:
print(classification_report(classe_teste, previsoes))

              precision    recall  f1-score   support

       <=50K       0.88      0.87      0.87      3693
        >50K       0.61      0.63      0.62      1192

    accuracy                           0.81      4885
   macro avg       0.74      0.75      0.75      4885
weighted avg       0.81      0.81      0.81      4885



### Resultados obtidos

O algoritmo possui 80.96% de acurácia. Não é um valor desejado de acurácia mas houve uma melhoria boa em relação ao algoritmo de Naive Bayes que possuia acurácia menor que 50%.